docs/management.yml

```diff
@@ -293,103 +267,63 @@ components:
+        #     priority:
+        #       type: number
+        #       format: integer
+        #     expression:
+        #       oneOf:
+        #         - '#/components/schemas/DeviceGroup'
+        provider:
+          type: string
+          enum:
+            - "iot-hub"
+        credentials:
+          oneOf:
+            - $ref: '#/components/schemas/SharedAccessSecret'
+            # - $ref: '#/components/schemas/x509Cert'
+          discriminator:
+            propertyName: type
+            mapping:
+              sas: '#/components/schemas/SharedAccessSecret'
+              # x509: '#/components/schemas/x509Cert'
+      required:
+        - provider
+        - credentials
 
-    DeviceTwin:
-      externalDocs:
-        url: >-
-          https://docs.microsoft.com/en-us/rest/api/iothub/service/devices/update-twin#twin
+    SharedAccessSecret:
       type: object
+      description: |
+        Shared Access Secret is an authentication mechanism in the form of a
+        connection string for Azure IoT Hub.
       properties:
-        authenticationType:
-          type: string
-          enum:
-            - certificateAuthority
-            - none
-            - sas
-            - selfSigned
-        capabilities:
-          type: object
-          description: Status of capabilities enabled on the device.
-          properties:
-            iotEdge:
-              type: boolean
-        cloudToDeviceMessageCount:
-          type: number
-          format: integer
-        connectionState:
-          type: string
-          enum: ["Connected", "Disconnected"]
-          description: The connection state of the device
-        deviceEtag:
-          type: string
-        deviceId:
-          type: string
-          description: The unique ID of the device within IoT Hub.
-        deviceScope:
-          type: string
-        etag:
-          type: string
-        lastActivityTime:
-          type: string
-          format: date-time
-        properties:
-          $ref: '#/components/schemas/TwinProperties'
-        status:
-          type: string
-          enum: ["enabled", "disabled"]
-          description: >-
-            The status of the device.
-            If disabled, the device cannot connect to the service.
-        statusReason:
-          type: string
-          description: Reason for last status change, if any.
-        statusUpdateTime:
+        connection_string:
           type: string
-          format: date-time
-        tags:
-          type: object
-          description: >-
-            User-defined key/value pairs assigned to the device.
-            The object is allowed to have a maximum depth of 10,
-            and must be less than 4KiB in size.
-          additionalProperties: true
-        version:
-          type: integer
-        X509Thumbprint:
-          $ref: '#/components/schemas/X509Thumbprint'
+      required: [connection_string]
 
-    TwinProperties:
+    DeviceState:
```

Comment:
What about the latest update timestamp?

Reply:
Should we include this right away? I'm not sure if it makes sense either, the Azure `LastActivityTime` we discussed yesterday refers to device's last connectivity time and not the Device Twin properties.

Reply:
Ok, then let's keep it out, thanks for the explanation!